.github/workflows/build-test.yaml

```diff
@@ -124,6 +124,21 @@ jobs:
             cargo build --target ${{ matrix.target }} --all-features
           fi
 
+  asan:
+    name: Address Sanitizer
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install upstream libseccomp
+        uses: ./.github/actions/setup
```

Comment:
Gould it be interesting to compile libseccomp with ASAN too?

Reply:
I think compiling libseccomp with ASAN should be done on the side of libseccomp community if it is needed.
The community have been using valgrind to detect memory errors, so asan is not always necessary. 
(But that's off topic, in general asan can detect more errors than valgrind..)

Besides, it wouldn't good our CI fails due to the side of the upstream libseccomp and our development stops. WDYT?

Reply:
We are not (primary) interested in libseccomp internal bugs. I was more  thinking about finding more on your side when we call a libseccomp function with e.g. a pointer that is invalid and gets dereferenced by libseccomp will the current ASAN detect it?

Reply:
Ah, exactly. if we accidentally pass invalid pointer to the libseccomp in our crate, the ASAN will be useful.
I'll try it out.

Reply:
I enabled ASAN, referring to https://wiki.tizen.org/ASan.

```sh
// Under the directory of the upstream libseccomp
$ ./configure --prefix=/usr/local/libseccomp CFLAGS="-O2 -g -fsanitize=address -U_FORTIFY_SOURCE -fno-omit-frame-pointer -fno-common"
$ make
$ make install

// Check ASAN is enabled in the libseccomp
$ nm /usr/local/libseccomp/lib/libseccomp.so | grep asan
                 U __asan_handle_no_return
                 U __asan_init
                 U __asan_option_detect_stack_use_after_return
                 U __asan_register_globals
                 U __asan_report_load1
                 U __asan_report_load16
                 U __asan_report_load2
                 U __asan_report_load4
                 U __asan_report_load8
                 U __asan_report_load_n
                 U __asan_report_store1
                 U __asan_report_store16
                 U __asan_report_store2
                 U __asan_report_store4
                 U __asan_report_store8
                 U __asan_report_store_n
                 U __asan_stack_free_6
                 U __asan_stack_malloc_0
                 U __asan_stack_malloc_1
                 U __asan_stack_malloc_2
                 U __asan_stack_malloc_3
                 U __asan_stack_malloc_6
                 U __asan_unregister_globals
                 U __asan_version_mismatch_check_v8
000000000003f7a5 b __odr_asan.arch_def_aarch64
000000000003f7a4 b __odr_asan.arch_def_arm
000000000003f7a7 b __odr_asan.arch_def_mips
000000000003f7a9 b __odr_asan.arch_def_mips64
000000000003f7ab b __odr_asan.arch_def_mips64n32
000000000003f7a6 b __odr_asan.arch_def_mipsel
000000000003f7a8 b __odr_asan.arch_def_mipsel64
000000000003f7aa b __odr_asan.arch_def_mipsel64n32
000000000003f7a0 b __odr_asan.arch_def_native
000000000003f7ac b __odr_asan.arch_def_parisc
000000000003f7ad b __odr_asan.arch_def_parisc64
000000000003f7ae b __odr_asan.arch_def_ppc
000000000003f7b0 b __odr_asan.arch_def_ppc64
000000000003f7af b __odr_asan.arch_def_ppc64le
000000000003f7b1 b __odr_asan.arch_def_riscv64
000000000003f7b2 b __odr_asan.arch_def_s390
000000000003f7b3 b __odr_asan.arch_def_s390x
000000000003f7a3 b __odr_asan.arch_def_x32
000000000003f7a1 b __odr_asan.arch_def_x86
000000000003f7a2 b __odr_asan.arch_def_x86_64
000000000003f741 b __odr_asan.library_version
000000000003f740 b __odr_asan.seccomp_api_level

// Under the directory of libseccomp-rs
$ export LIBSECCOMP_LIB_PATH=/usr/local/libseccomp/lib
$ export LD_LIBRARY_PATH=$LIBSECCOMP_LIB_PATH  
$ RUSTFLAGS="-Z sanitizer=address" cargo +nightly test --target x86_64-unknown-linux-gnu --tests -- --color always --nocapture --test-threads 1
    Finished test [unoptimized + debuginfo] target(s) in 0.01s
     Running unittests src/lib.rs (target/x86_64-unknown-linux-gnu/debug/deps/libseccomp-e3e3f3de5a7e05e7)
==4103023==Your application is linked against incompatible ASan runtimes.
error: test failed, to rerun pass `-p libseccomp --lib`

Caused by:
  process didn't exit successfully: `***/libseccomp-rs/target/x86_64-unknown-linux-gnu/debug/deps/libseccomp-e3e3f3de5a7e05e7 --color always --nocapture --test-threads 1` (exit status: 1)
```
Got the error, **Your application is linked against incompatible ASan runtimes.**
I found the same issues:
https://users.rust-lang.org/t/asan-across-c-dynamic-library-and-rust-crate/96649
https://github.com/rust-lang/rust/issues/114127

Then I added `-static-libasan` option to GCC and confirmed that `cargo test` with ASAN worked.

```sh
$ ./configure --prefix=/usr/local/libseccomp CFLAGS="-O2 -g -fsanitize=address -static-libasan -U_FORTIFY_SOURCE -fno-omit-frame-pointer -fno-common"
// Do same the above
...
$ RUSTFLAGS="-Z sanitizer=address" cargo +nightly test --target x86_64-unknown-linux-gnu --tests -- --color always --nocapture --test-threads 1
 x86_64-unknown-linux-gnu --tests -- --color always --nocapture --test-threads 1
    Finished test [unoptimized + debuginfo] target(s) in 0.00s
     Running unittests src/lib.rs (target/x86_64-unknown-linux-gnu/debug/deps/libseccomp-e3e3f3de5a7e05e7)

running 17 tests
...
```

From the https://github.com/rust-lang/rust/issues/114127:
>The cause:
When using RUSTFLAGS=-Zsanitizer=address, it seems rustc always static links asan runtime,
by static link librustc-nightly_rt.asan.a bundled with rustc installation
I think static links libasan, is not compatible with dynamic linked libasan
I don't think rustc provides a way to opt out of static linking librustc-nightly_rt.asan.a

Hence, we have to link `libasan` statically in order to run tests for the libseccomp with ASAN enabled.
@rusty-snake if there is no problem, I'll update this PR to enable ASAN for the libseccomp.

Reply:
Even tough the libseccomp links libasan statically, the symbols `__asan_*` seem to be undefinied.
I'm not sure this is a correct behavior, so I'm now taking a look at the related issue:  https://github.com/google/sanitizers/issues/1086

```sh
$nm /usr/local/libseccomp/lib/libseccomp.so | grep asan
                 U __asan_handle_no_return
                 U __asan_init
                 U __asan_option_detect_stack_use_after_return
                 U __asan_register_globals
                 U __asan_report_load1
                 U __asan_report_load16
                 U __asan_report_load2
                 U __asan_report_load4
                 U __asan_report_load8
                 U __asan_report_load_n
                 U __asan_report_store1
                 U __asan_report_store16
                 U __asan_report_store2
                 U __asan_report_store4
                 U __asan_report_store8
                 U __asan_report_store_n
                 U __asan_stack_free_6
                 U __asan_stack_malloc_0
                 U __asan_stack_malloc_1
                 U __asan_stack_malloc_2
                 U __asan_stack_malloc_3
                 U __asan_stack_malloc_6
                 U __asan_unregister_globals
                 U __asan_version_mismatch_check_v8
000000000003f7a5 b __odr_asan.arch_def_aarch64
000000000003f7a4 b __odr_asan.arch_def_arm
000000000003f7a7 b __odr_asan.arch_def_mips
000000000003f7a9 b __odr_asan.arch_def_mips64
000000000003f7ab b __odr_asan.arch_def_mips64n32
000000000003f7a6 b __odr_asan.arch_def_mipsel
000000000003f7a8 b __odr_asan.arch_def_mipsel64
000000000003f7aa b __odr_asan.arch_def_mipsel64n32
000000000003f7a0 b __odr_asan.arch_def_native
000000000003f7ac b __odr_asan.arch_def_parisc
000000000003f7ad b __odr_asan.arch_def_parisc64
000000000003f7ae b __odr_asan.arch_def_ppc
000000000003f7b0 b __odr_asan.arch_def_ppc64
000000000003f7af b __odr_asan.arch_def_ppc64le
000000000003f7b1 b __odr_asan.arch_def_riscv64
000000000003f7b2 b __odr_asan.arch_def_s390
000000000003f7b3 b __odr_asan.arch_def_s390x
000000000003f7a3 b __odr_asan.arch_def_x32
000000000003f7a1 b __odr_asan.arch_def_x86
000000000003f7a2 b __odr_asan.arch_def_x86_64
000000000003f741 b __odr_asan.library_version
000000000003f740 b __odr_asan.seccomp_api_level
```

Reply:
The above undefined symbol seems to be expected behavior even `-static-libasan` is specified. The symbol is resolved at runtime when the executable binary runs.